de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,272 @@
+    }
+
+    /**
+     * Extract Parameter
+     */
+    private def generate(Parameter parameter) {
+    	// Formatting is kinda ugly because otherwise xtend's string 
+    	// templates have additional not intended linebreaks
+        return '''«
+        FOR a : parameter.aspects SEPARATOR ' '
+        	»«a.generate»«
+        ENDFOR
+        » «parameter.communicationType.generate» «parameter.exchangePattern.generate» «
+        parameter.name» : «parameter.generateType
+        »'''
+    }
+    
+    /**
+     * Extract CommunicationType
+     */
+    private def generate(CommunicationType type) {
+        return switch(type) {
+            case ASYNCHRONOUS: "async"
+            case SYNCHRONOUS: "sync"
+            default: null
+        }
+    }
+
+    /**
+     * Extract ExchangePattern
+     */
+    private def generate(ExchangePattern pattern) {
+        return switch(pattern) {
+            case IN: "in"
+            case OUT: "out"
+            case INOUT: "inout"
+            default: null
+        }
+    }
+
+    /**
+     * Extract ImportedServiceAspect
+     */
+    private def generate(ImportedServiceAspect aspect) {
+        return '''@«aspect.importedAspect.technology.name»::«
+        FOR s : aspect.importedAspect.getQualifiedNameParts(false, true) SEPARATOR '.'»«s
+        »«ENDFOR»'''
+    }
+
+    /**
+     * Extract Parameter
+     */
+    private def generateType(Parameter parameter) {
+        return if (parameter.primitiveType !== null)
+                (parameter.primitiveType as PrimitiveType).generate
+            else
+                parameter.importedType.generate
+    }
+
+    /**
+     * Extract PrimitiveType
+     */
+    private def generate(PrimitiveType type) {
+        return type.typeName
+    }
+    
+    /**
+     * Extract Technology Annotations of a Microservice
+     */   
+    private def generateTechAnnotation(Microservice service) {
+        return '''«FOR tech :
+            service.serviceModel.imports.filter[it.importType == ImportType.TECHNOLOGY]
+            »@technology(«tech.name»)«ENDFOR»'''
+    }
+    
+    /**
+     * Extract Protocol
+     */    
+    private def generate(Protocol protocol) {
+        val techName = protocol.technology?.name ?: '''[PROTOCOL TECHNOLOGY URI NOT DEFINED]'''
+        return '''«techName»::«FOR p : protocol.qualifiedNameParts SEPARATOR '.'»«p»«ENDFOR»'''
+    }
+    
+    /**
+     * Extract ImportedType
+     */ 
+    private def generate(ImportedType importedType) {
+        return switch (importedType.import.importType) {
+            case ImportType.TECHNOLOGY:
+            	'''«importedType.import.name»::«importedType.type»'''
+            case ImportType.DATATYPES: {
+                val importedTypeName = (importedType.type as ComplexType).buildQualifiedName(".")
+                '''«importedType.import.name»::«importedTypeName»'''
+            }
+            default:
+                throw new UnsupportedTypeException(
```

Comment:
Throw an `IllegalArgumentException here` instead of defining a new `Exception` for only this particular edge-case.